### ISSUE TITLE
`cast_possible_wrap` for convertion from `u8` to `isize`

### ISSUE THREAD

**Issue by quarthex** [rust-lang/rust-clippy#9337]
### Summary

The lint `cast_possible_wrap` is triggered by a conversion from `u8` to `isize`.

The displayed message is:

    casting `u8` to `isize` may wrap around the value on targets with 32-bit wide pointers

On 16/32/64-bits platforms, the message is wrong. An 8-bit unsigned value fits in a 16/32/64-bits signed register.

### Reproducer

```rust
#![warn(clippy::cast_possible_wrap)]
const A: u8 = 0;
const B: isize = A as isize;
```

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

### Additional Labels

@rustbot label +I-false-positive +L-pedantic

**Comment by quarthex** [comment]
The lint applies on 8-bits platforms, though.

**Comment by asquared31415** [comment]
Rust will never support platforms where `u/isize` is smaller than 16 bits and this lint is always incorrect.  In fact `isize::from(u8)` exists, which means it definitely is a lossless conversion.

**Comment by quarthex** [comment]
@asquared31415 If it want to be considered as a language targeted for embedded development, Rust _should_ be able to produce code for such platforms. In my company, we are trying to deal with shortages of microcontrollers such as STM32 (mainly reserved by the car industry) and we will use TinyAVR micrcontrollers instead.
So, saying that Rust will _never_ support 8-bit platforms is a little dramatic from my point of view.

**Comment by asquared31415** [comment]
Unfortunately Rust has already made guarantees that are incompatible with that.  It's well documented that `usize` is at least 16 bits large in [The Reference](https://doc.rust-lang.org/reference/types/numeric.html#machine-dependent-integer-types) and changing that would also require breaking changes to the standard library, namely the removal of the `isize::from(u8)` impl.
